### ISSUE TITLE
observer should run in a separate process

### ISSUE THREAD

**Issue by britram** [mami-project/pathspider#28]
The observer is a relatively heavyweight thread and starves other Python threads from running (see #25). On multicore machines (or VMs), this can be fixed by running the observer in a separate process. Consider multiprocessing or concurrent.futures for this.

**Comment by britram** [comment]
`multiprocessing` is probably the right way to go, with the caveat that the observer needs to start before *any* of the other threads do.

**Comment by britram** [comment]
This is in the `multiprocessing-observer` branch. Current bug: mp.Queue doesn't support task_done:

```
Traceback (most recent call last):
  File "/home/vagrant/pathspider/pathspider/base.py", line 331, in exception_wrapper
    target(*args, **kwargs)
  File "/home/vagrant/pathspider/pathspider/base.py", line 303, in merger
    self.flowqueue.task_done()
AttributeError: 'Queue' object has no attribute 'task_done'
```

**Comment by britram** [comment]
Shutdown is clean now, but still needs final flush / wait for end: running ecnspider with 12 workers on 100 inputs only results in 32 results.

**Comment by britram** [comment]
Fixed by 58d46a5ede8dd2ca986328668a758f1048b5d122, awaiting merge to master

**Comment by irl** [comment]
This is merged to master now.
